src/mscorlib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -450,21 +429,6 @@ public void SetStateMachine(IAsyncStateMachine stateMachine)
                 TaskAwaiter.UnsafeOnCompletedInternal(vta.AsTask(), box, vta._continueOnCapturedContext);
             }
 
-            // To catch all Task/Task<T> awaits, do the currently more expensive interface checks.
-            // Eventually these and the above Task/Task<T> checks should be replaced by "is" checks,
-            // once that's recognized and optimized by the JIT.  We do these after all of the hardcoded
-            // checks above so that they don't incur the costs of these checks.
-            else if (InterfaceIsCheckWorkaround<TAwaiter>.IsITaskAwaiter)
-            {
-                ref TaskAwaiter ta = ref Unsafe.As<TAwaiter, TaskAwaiter>(ref awaiter);
-                TaskAwaiter.UnsafeOnCompletedInternal(ta.m_task, box, continueOnCapturedContext: true);
-            }
-            else if (InterfaceIsCheckWorkaround<TAwaiter>.IsIConfiguredTaskAwaiter)
```

Comment:
The InterfaceIsCheckWorkaround class could then also be deleted.

Reply:
It's gone now.